tests/test_fsspec.py

```diff
@@ -0,0 +1,59 @@
+"""
+Test ability to index off diverse filesystems using fsspec.
+"""
+
+import json
+
+import fsspec.core
+import pytest
+
+from conda_index.index import ChannelIndex
+from conda_index.index.fs import FsspecFS
+
+
+def test_fsspec(tmp_path, http_package_server):
+    """
+    Test that conda-index can directly index remote files.
+    """
+    channel_root = tmp_path / "channel"  # used for sqlite cache in this mode
+    channel_root.mkdir()
+    output = tmp_path / "output"  # default same as channel
+    output.mkdir()
+
+    host, port = http_package_server.socket.getsockname()
+    channel_url = f"http://{host}:{port}/"
+
+    # e.g. http://127.0.0.1:54963/osx-64/
+
+    test_subdirs = ("noarch", "osx-64")
```

Comment:
worth parameterizing it?

Reply:
these are the extant subdirectories in index_data/packages/